### PR TITLE
Attention net body protobuf update with smolgen weights

### PR DESCRIPTION
**Pull request by almaudoh**
Protobuf updates for attention body with support for smolgen weights.

### CODE REVIEW

File: proto/net.proto [hunk 1]
```diff
@@ -198,12 +232,37 @@ message NetworkFormat {
   }
   optional MovesLeftFormat moves_left = 6;
 
+  enum ActivationFunction {
+    DEFAULT = 0;
```
Comment (Tilps) [inline proto/net.proto:236]: Could call this inherit.

File: proto/net.proto [hunk 2]
```diff
@@ -98,6 +111,18 @@ message Weights {
   // Residual tower.
   repeated Residual residual = 2;
 
+  // Embedding layer for attention body encoders.
```
Comment (Tilps) [inline proto/net.proto:114]: Maybe good if you add comments against new things that will only be interpreted if network type is attentionbody_with_headformat, that that is the case.

File: proto/net.proto (hunk 1 above)
Comment (borg323) [reply proto/net.proto:236]: My thinking was that inherit is the default action, unless we specify some other default, so this is a bit more flexible. Originally I defined AttentionPolicyActivation, with SELU as default to cover T78-T81 net behavior, but in the end decided that it is better to handle this in the backends with a `NETWORK_SE_WITH_HEADFORMAT` check. Following this, `INHERIT` at this point felt a bit awkward, so I decided to not revert it.

Comment (Tilps) [reply proto/net.proto:236]: based on the comment attached to default activation - we can put a comment in the attention policy section that 'for network type prior to NETWORK_ATTENTIONBODY_WITH_HEADFORMAT' FFN activation in these encoders is SELU - afterwards it follows standard ffn activation function setting.

Comment (Tilps) [reply proto/net.proto:236]: Sounds about right.

File: proto/net.proto (hunk 2 above)
Comment (almaudoh) [reply proto/net.proto]: Comments added with @borg323's PR.

File: proto/net.proto [hunk 3]
```diff
@@ -198,12 +238,37 @@ message NetworkFormat {
   }
   optional MovesLeftFormat moves_left = 6;
 
+  enum ActivationFunction {
+    DEFAULT = 0;
```
Comment (mooskagh) [inline proto/net.proto:242]: As per style guide (as enums polute parent namespaces),

DEFAULT_ACTIVATION,
ACTIVATION_MISH,
ACTIVATION_RELU
etc

File: proto/net.proto [hunk 4]
```diff
@@ -198,12 +238,37 @@ message NetworkFormat {
...
+
   // Activation used everywhere except head outputs or otherwise specified.
   enum DefaultActivation {
     DEFAULT_ACTIVATION_RELU = 0;
     DEFAULT_ACTIVATION_MISH = 1;
   }
   optional DefaultActivation default_activation = 7;
+
+  enum SmolgenActivation {
+    SMOLGEN_ACTIVATION_INHERIT = 0;
```
Comment (mooskagh) [inline proto/net.proto:262]: As discussed in Discord, I'd prefer to reuse the same enum for activations in all places, but I'm fine with separate enums if other people feel strongly about it.

File: proto/net.proto (hunk 3 above)
Comment (almaudoh) [reply proto/net.proto]: Done

File: proto/net.proto (hunk 4 above)
Comment (almaudoh) [reply proto/net.proto]: Done

**Review by Tilps — APPROVED** [review]

**Pull request merged**